rvt-patch-llvm/src/main.rs

```diff
@@ -261,6 +283,70 @@ fn insert_call_at_head<'a>(
     builder.build_call(f, &args, "");
 }
 
+////////////////////////////////////////////////////////////////
+// Function transformation functions
+////////////////////////////////////////////////////////////////
+
+/// Apply Rust mangling rule to a function name like 'foo::bar'.
+/// Does not insert a unique hash at the end or the final 'E' character.
+fn rust_mangle(n: &str) -> String {
+    // todo: I believe that this can be implemented by an elegant chain of
+    // iterators - but I am not smart enough to figure out the ownership, etc.
+    // issues that rustc reports so I went with the following code instead.
+    let mut rs = vec!["_ZN".to_string()];
+    for x in n.split("::") {
+        rs.push(format!("{}{}", x.len(), x));
+    }
+    rs.concat()
+}
+
+/// Find a function whose name matches a Rust-mangled prefix
+fn find_mangled_function<'a>(module: &Module<'a>, prefix: &str) -> Option<FunctionValue<'a>> {
```

Comment:
There is somewhat similar function already, `get_function`, we should probably unify them.

Reply:
I refactored these into get_function, get_function_by_regex and get_function_by_unmangled_name